core/src/main/java/org/elasticsearch/search/suggest/SuggestBuilder.java

```diff
@@ -212,5 +227,68 @@ public T shardSize(Integer shardSize) {
             this.shardSize = shardSize;
             return (T)this;
         }
+
+        @Override
+        public String getWriteableName() {
+            return name;
```

Comment:
I'm not 100% familiar with suggesters at this point, but from looking at the docs, I think `name` here refers to some custom name for each suggest entry in the request (like "my-location-suggestion" etc..) and its used amongst other things to name the results in the response. The NamedWritable#getWritbaleName() should provide a unique name for the type of writable object. In this case I think the abstract class shouldn't implement this, but the three implementations (Suggest/Term/Completion) whould return how they are identified here.